cli/src/commands/rebase.rs

```diff
@@ -481,28 +475,27 @@ fn rebase_descendants_transaction(
     tx.finish(ui, tx_description)
 }
 
-/// Computes the new parents and children given the input arguments for
-/// `destination`, `insert_after`, and `insert_before`.
+/// Computes the new parents and children for the given
+/// [`RebaseDestinationArgs`].
 fn compute_rebase_destination(
     ui: &mut Ui,
     workspace_command: &mut WorkspaceCommandHelper,
-    destination: &[RevisionArg],
-    insert_after: &[RevisionArg],
-    insert_before: &[RevisionArg],
+    rebase_destination: &RebaseDestinationArgs,
 ) -> Result<(Vec<Commit>, Vec<Commit>), CommandError> {
-    let resolve_revisions = |revisions: &[RevisionArg]| -> Result<Vec<Commit>, CommandError> {
-        if revisions.is_empty() {
-            Ok(vec![])
-        } else {
-            Ok(workspace_command
-                .resolve_some_revsets_default_single(ui, revisions)?
-                .into_iter()
-                .collect_vec())
-        }
-    };
-    let destination_commits = resolve_revisions(destination)?;
-    let after_commits = resolve_revisions(insert_after)?;
-    let before_commits = resolve_revisions(insert_before)?;
+    let resolve_revisions =
+        |revisions: &Option<Vec<RevisionArg>>| -> Result<Vec<Commit>, CommandError> {
+            if let Some(revisions) = revisions {
+                Ok(workspace_command
+                    .resolve_some_revsets_default_single(ui, revisions)?
+                    .into_iter()
+                    .collect_vec())
+            } else {
+                Ok(vec![])
+            }
+        };
+    let destination_commits = resolve_revisions(&rebase_destination.destination)?;
+    let after_commits = resolve_revisions(&rebase_destination.insert_after)?;
+    let before_commits = resolve_revisions(&rebase_destination.insert_before)?;
```

Comment:
nit: I meant these `Option<&[RevisionArg]>`s can be mapped to `Option<Vec<Commit>>`s so the following `if` block can be a single `match` statement.